workshops/2021-16-02_intro_to_github/TOG_git_tutorial.Rmd

```diff
@@ -0,0 +1,396 @@
+
+Before we can start using Git, your computer must be equipped with the appropriate softwares. In this section, we will walk you through: 1. Installing Git, 2. Registering for a GitHub account, 3. Setting up an SSH key, 4. Talking to Git via RStudio. 
+
+### 1. Install Git (contains excerpts from [Ch. 7 of Happy Git and GitHub for the useR](http://happygitwithr.com/install-git.html))
+
+You need Git, so you can use it at the command line and so RStudio can call it. If there’s any chance it’s installed already, verify that, rejoice, and skip this step. Otherwise, find installation instructions below for your operating system.
+
+#### Git already installed?
+
+Open the shell (terminal / command line). Type in "git --version", no quotation, press enter.
+
+```{r eval = FALSE}
+git --version
+## git version 2.11.0 (Apple Git-81)
+```
+
+If you are successful, that’s great! You have Git already. No need to install! Move on.
+
+If, instead, you see something more like git: command not found, keep reading.
+
+#### Install Git for Windows people
+
+Install [Git for Windows](https://git-for-windows.github.io/), previously known as msysgit or “Git Bash”, to get Git in addition to some other useful tools, such as the Bash shell. Yes, all those names are totally confusing. You may accept all the default settings during installation.
+
+* This approach leaves the Git executable in a conventional location, which will help you and other programs, e.g. RStudio, find it and use it. This also supports a transition to more expert use, because the Bash shell will be useful as you venture outside of R/RStudio.
+* This also leaves you with a Git client, though not a very good one. 
+* RStudio for windows likes for git to be in the Files(x86) folder. If not in this location, RStudio may not detect it, and may cause headaches for you later.
+
+
+#### Install Git for Mac people
+
+Install the Xcode command line tools (not all of Xcode), which includes Git by opening a terminal and typing in the following and press enter. Follow prompts. 
+
+```{r eval = FALSE}
+xcode-select --install
+```
+
+Verify that Xcode command line tools has been installed by typing the following into the terminal and press enter.
+
+```{r eval = FALSE}
+xcode-select --version
+## xcode-select version 2347.
+```
+
+Enter one of these commands to elicit an offer to install developer command line tools:
+
+```{r eval = FALSE}
+git --version
+git config
+```
+
+Accept the offer! Click on “Install”.
+
+#### Install Git for Linux people
+
+Install Git via your distro’s package manager.
+
+Ubuntu or Debian Linux:
+
+```{r eval = FALSE}
+sudo apt-get install git
+```
+
+Fedora or RedHat Linux:
+
+```{r eval = FALSE}
+sudo yum install git
+```
+
+[A comprehensive list](https://git-scm.com/download/linux) for various Linux and Unix package managers.
+
+
+### 2. Register for a GitHub account (contains excerpts from [Ch. 5 of Happy Git and GitHub for the useR](http://happygitwithr.com/install-git.html))
```

Comment:
Perhaps suggest that people do this before coming, just the account creation part